src/Acuminator/Acuminator.Analyzers/StaticAnalysis/NonPublicExtensions/NonPublicExtensionFix.cs

```diff
@@ -0,0 +1,102 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Acuminator.Utilities.Common;
+using Acuminator.Utilities.Roslyn.Semantic;
+using Acuminator.Utilities.Roslyn.Syntax;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Acuminator.Analyzers.StaticAnalysis.NonPublicExtensions
+{
+	[Shared]
+	[ExportCodeFixProvider(LanguageNames.CSharp)]
+	public class NonPublicExtensionFix : CodeFixProvider
+	{
+		private const string DiagnosticID = "PX1022";
```

Comment:
@dropsonic  I deliberately made it this way because if you write only one of the descriptors it leaves an impression that fix only works for it. 
I can use Distinct with both descriptors, though it will be an overhead

Reply:
@dropsonic I deliberately made it this way because if you write only one of the descriptors it leaves an impression that fix only works for it.
I can use Distinct with both descriptors, though it will be an overhead

Reply:
Imho, it seems reasonable to use `Distinct` because we had a couple of cases where we changed IDs for existing diagnostics. It is more clear than having IDs hardcoded.

Reply:
ok, I'll change it

Reply:
Done